lib/fizzy/parser_expr.cpp

```diff
@@ -295,12 +305,12 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, const Mod
             std::tie(arity, pos) = parse_blocktype(pos, end);
             code.immediates.push_back(arity);
 
+            // Push label with immediates offset after arity.
+            control_stack.push({Instr::block, frame.stack_height, code.immediates.size()});
```

Comment:
This `push` can theoretically invalidate `frame` reference, so `frame` should be updated before.

Reply:
ah this is changed in later commit

Reply:
Oops, I have not see this. In this case this is fine as we use `std::deque`, but still I should have think about this before :) I guess I need some tests that causes stack memory grow here and then ASan should find the bug, if it ever happen.

Reply:
https://github.com/wasmx/fizzy/issues/321

Reply:
Yeah I think I saw this happening in #312 (which uses `Stack` based on `vector`), that's why I noticed. Kind of subtle confusing error.

Reply:
Let me know if you have any repro tests. I will try to handle that on the current master.